src/Symfony/Component/Notifier/Transport/Dsn.php

```diff
@@ -25,9 +25,9 @@ final class Dsn
     private $port;
     private $options;
     private $path;
-    private $dsn;
+    private $originalDsn;
 
-    public function __construct(string $scheme, string $host, ?string $user = null, ?string $password = null, ?int $port = null, array $options = [], ?string $path = null)
+    private function __construct(string $scheme, string $host, ?string $user = null, ?string $password = null, ?int $port = null, array $options = [], ?string $path = null)
```

Comment:
I don't see why we should forbid the usage of the constructor.

Reply:
Otherwise you would end up with the same problem.... original DSN is not set, but used heavily in exceptions.

Reply:
Instead, would it possible to create the "DSN as string" from all the parts given in the constructor?

Reply:
Yes it would, but it is not the "original"/"provided" one, so let's say we can't parse anything or have magic chars or .... we will not use the provided one in the exception message, but a generated one which is not what I would expect 🤷‍♂️ 

So this would be more a "toString" instead of a "original dsn"...

Besides this, its a class to deal with DSN's which are strings per design, so I would go with strings only, another positive aspect I encountered while reworking all tests for the bridges: the dsn is much more readable as a string ❗

Reply:
I get it now :) So, what about removing the fromString() method instead and pass the DSN (a string) as the only argument to the constructor?

Reply:
@OskarStark Have you seen my comment above?

Reply:
Didn't see the comment, will have a look later or tomorrow 👍🏻

Reply:
> I get it now :) So, what about removing the fromString() method instead and pass the DSN (a string) as the only argument to the constructor?

Yes we can 👍  Will do